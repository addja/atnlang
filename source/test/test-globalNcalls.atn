b = "adeu! %n";
c = 22;

def printer(a) {
	print a;
	print b;
	print c, "%n";
	c = c+23;
	print c, "%n";
}

def main() {
	printer("hola! %n");
	print c, "%n";
}
