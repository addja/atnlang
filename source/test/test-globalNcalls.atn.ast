(PROGRAM (:= b "adeu! %n") (:= c 22) (def printer (PARAMS a) (LIST_INSTR (print a) (print b) (print c))) (def main PARAMS (LIST_INSTR (FUNCALL printer (ARGLIST "hola! %n")))))