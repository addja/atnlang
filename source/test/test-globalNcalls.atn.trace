main() <entry point>
|   printer(a=hola! %n) <line 13>
|   return <line 9>
return <line 14>
