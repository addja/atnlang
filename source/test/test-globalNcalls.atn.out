hola! 
adeu! 
22
45
45
